### ISSUE TITLE
Open Dict Type

### ISSUE THREAD

**Issue by ambs** [TLINDEN/Data-Validate-Struct#5]
In a webservice I am developing, it might make sense for one parameter to be a open dictionary (hash), with just a constrain on the key / value.

Something like:

```
  $model = { id => 'int', metadata => 'dict(word, line)' };
```

This idea is still not well thought, but I decided to share it, so you can give also some input.

**Comment by TLINDEN** [comment]
Sounds like a good idea, I thought about implementing this validation with the current module version, but this is not possible. Also, variations of this come to mind like `array(int)` or even `array(dict(word, array(word, line)))`.

**Comment by ambs** [comment]
Yes, the current reference structure is quite nice, because it is simple.
I looked up other modules like JSON::Schema::Modern, but the reference definition is too complex for what I need.

Thus, any change here needs to be taken with caution to keep the module simple :-)
